api_tests/collections/test_views.py

```diff
@@ -222,2261 +234,2617 @@ def test_create_bookmark_collection_with_wrong_title(self):
-            expect_errors=True
-        )
-        assert_equal(res.status_code, 404)
+            auth=user_one.auth,
+            expect_errors=True)
+        assert res.status_code == 404
 
+@pytest.mark.django_db
+class CollectionCRUDTestCase:
 
-class CollectionCRUDTestCase(ApiTestCase):
+    @pytest.fixture()
+    def user_two(self):
+        return AuthUserFactory()
 
-    def setUp(self):
-        super(CollectionCRUDTestCase, self).setUp()
-        self.user = AuthUserFactory()
-        self.user_two = AuthUserFactory()
+    @pytest.fixture()
+    def title_collection(self):
+        return 'Cool Collection'
 
-        self.title = 'Cool Collection'
-        self.new_title = 'Super Cool Collection'
+    @pytest.fixture()
+    def new_title_collection(self):
+        return 'Super Cool Collection'
 
-        self.collection = CollectionFactory(title=self.title, creator=self.user)
-        self.url = '/{}collections/{}/'.format(API_BASE, self.collection._id)
-        self.fake_url = '/{}collections/{}/'.format(API_BASE, '12345')
+    @pytest.fixture()
+    def collection(self, title_collection, user_one):
+        return CollectionFactory(title=title_collection, creator=user_one)
 
+    @pytest.fixture()
+    def url_collection_detail(self, collection):
+        return '/{}collections/{}/'.format(API_BASE, collection._id)
 
-def make_collection_payload(collection, attributes):
-    return {
-        'data': {
-            'id': collection._id,
-            'type': 'collections',
-            'attributes': attributes,
-        }
-    }
+    @pytest.fixture()
+    def url_fake_collection_detail(self, collection):
+        return '/{}collections/{}/'.format(API_BASE, '12345')
 
+    @pytest.fixture()
+    def payload_collection(self, collection):
+        def make_collection_payload(attributes):
+            return {
+                'data': {
+                    'id': collection._id,
+                    'type': 'collections',
+                    'attributes': attributes,
+                }
+            }
+        return make_collection_payload
 
+@pytest.mark.django_db
 class TestCollectionUpdate(CollectionCRUDTestCase):
 
-    def test_node_update_invalid_data(self):
-        res = self.app.put_json_api(self.url, "Incorrect data", auth=self.user.auth, expect_errors=True)
-        assert_equal(res.status_code, 400)
-        assert_equal(res.json['errors'][0]['detail'], "Malformed request.")
+    def test_update_collection_logged_in(self, app, url_collection_detail, collection, new_title_collection, user_one):
+        with assert_latest_log(NodeLog.EDITED_TITLE, collection):
+            res = app.put_json_api(url_collection_detail, {
+                'data': {
+                    'id': collection._id,
+                    'type': 'collections',
+                    'attributes': {
+                        'title': new_title_collection,
+                    }
+                }
+            }, auth=user_one.auth)
```

Comment:
Minor: Only the above line needs to be in the context manager. Better to make the `with` block as constrained as possible. Please change this in the other usages as well.